docs/topics/serialization.md

```diff
@@ -0,0 +1,78 @@
+[//]: # (title: Serialization)
+
+**Serialization** is the process of converting data used by an application to a format that can be transferred over a
+network or stored in a database or a file. Deserialization is the opposite process of converting external data back into a runtime object.
+Together, they are essential to most applications that exchange data with third parties.
+
+Some data serialization formats, such as [JSON](https://www.json.org/json-en.html) and [Protocol Buffers](https://protobuf.dev/) are particularly common.
+Being language-neutral and platform-neutral, these formats enable data exchange between systems written in any modern language.
+
+To convert an object tree to a string or to a sequence of bytes, it must go through two mutually intertwined processes: 
+
+1. Serialization: Objects are transformed into a sequence of their primitive values.
+This universal process varies depending on the object and is managed by a serializer.
+2. Encoding: The primitive sequence is converted into the desired output format, controlled by an encoder.
+
+![Serialization flow](serialization.svg){width=700}
+
+The reverse process involves parsing the input format, decoding the primitive values, and then deserializing the resulting
+stream into objects.
+
+If you're new to serialization in Kotlin, we recommend starting with the [Get Started with Serialization](serialization-get-started.md) guide.
+This section provides a step-by-step guide to help you set up and use Kotlin serialization in your projects.
+By following these steps, you can quickly get up to speed with the basics before diving into more complex topics.
+
+## Kotlin serialization libraries
+
+The `kotlinx.serialization` library offers support for all platforms, including JVM, JavaScript, Native.
```

Comment:
It may be worthwhile to clarify more that the serialization library provides the format agnostic infrastructure. Then the `-json` and `-cbor` libraries are format implementations (they are fundamentally different and are independent even though first-party).